wgpu/src/backend/wgpu_core.rs

```diff
@@ -437,133 +436,361 @@ fn map_pass_channel<V: Copy + Default>(
+            // Same goes for native shader module generation errors.
+            CreateShaderModuleError::Device(_) | CreateShaderModuleError::Generation => {
+                CompilationInfo {
+                    messages: Vec::new(),
+                }
+            }
+            // Everything else is an error message without location information.
+            _ => CompilationInfo {
+                messages: vec![CompilationMessage {
+                    message: value.to_string(),
+                    message_type: CompilationMessageType::Error,
+                    location: None,
+                }],
+            },
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct CoreQueueWriteBuffer {
+    buffer_id: wgc::id::StagingBufferId,
+    mapping: CoreBufferMappedRange,
+}
+
+#[derive(Debug)]
+pub struct CoreBufferMappedRange {
+    ptr: NonNull<u8>,
+    size: usize,
+}
+
+#[cfg(send_sync)]
+unsafe impl Send for CoreBufferMappedRange {}
+#[cfg(send_sync)]
+unsafe impl Sync for CoreBufferMappedRange {}
+
+impl Drop for CoreBufferMappedRange {
+    fn drop(&mut self) {
+        // Intentionally left blank so that `BufferMappedRange` still
+        // implements `Drop`, to match the web backend
+    }
+}
+
+crate::cmp::impl_eq_ord_hash_arc_address!(ContextWgpuCore => .0);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreAdapter => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreDevice => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreQueue => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreShaderModule => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreBindGroupLayout => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreBindGroup => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreTextureView => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreSampler => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreBuffer => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreTexture => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreBlas => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreTlas => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreQuerySet => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CorePipelineLayout => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreRenderPipeline => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreComputePipeline => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CorePipelineCache => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreCommandEncoder => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreComputePass => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreRenderPass => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreCommandBuffer => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreRenderBundleEncoder => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreRenderBundle => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreSurface => .id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreSurfaceOutputDetail => .surface_id);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreQueueWriteBuffer => .mapping.ptr);
+crate::cmp::impl_eq_ord_hash_proxy!(CoreBufferMappedRange => .ptr);
```

Comment:
Lookie: Now the backend types are responsible for implementing PartialEq etc how they see fit. Most of them use this proxy type, which just uses a single member for the partial eq behavior.